lib/jsdom/living/helpers/http-request.js

```diff
@@ -97,9 +97,18 @@ module.exports = class Request extends Writable {
     const scheme = urlOptions.protocol;
     this._requestOptions.agent = this.agents[scheme.substring(0, scheme.length - 1)];
     const { request } = scheme === "https:" ? https : http;
-    this._currentRequest = request(this.currentURL, this._requestOptions, response => {
-      this._processResponse(response);
-    });
+    if (http.request.length === 2) {
+      const { href, origin, protocol, username, password, host, hostname, port, pathname, search } = urlOptions;
+      const urlOptionsClone =
+        { href, origin, protocol, username, password, host, hostname, port, pathname, search, path: pathname };
+      this._currentRequest = request({ ...urlOptionsClone, ...this._requestOptions }, response => {
+        this._processResponse(response);
+      });
```

Comment:
The following version is simpler and also passes the tests:

```suggestion
      this._currentRequest = request({ ...urlOptions, ...this._requestOptions }, response => {
        this._processResponse(response);
      });
```

Reply:
The spread operator only spreads own propertise, but href, origin, protocol, etc. all are setters and getters of prototype of urlOptions. { ...urlOptions } produces empty object. I fixed the bug of assert.isRejected and added test case that the simpler version should fail.